src/com/google/enterprise/adaptor/fs/FsAdaptor.java

```diff
@@ -700,7 +714,9 @@ private void validateShare(Path sharePath) throws IOException {
     // Verify that the adaptor has permission to read the Acl and share Acl.
     try {
       readShareAcls(sharePath);
-      delegate.getAclViews(sharePath);
+      if (delegate.isDfsNamespace(sharePath) && allowFilesInDfsNamespaces) {
```

Comment:
I don't understand this condition. Previously, when isDfsNamespace was true, we didn't get here at all. Should this be !isDfsNamespace || allowFiles?